frontend/packages/ux-editor/src/components/config/editModal/EditOptions/OptionTabs/EditTab/EditTab.tsx

```diff
@@ -1,112 +1,69 @@
-  const [componentHasOptionList, setComponentHasOptionList] = useState<boolean>(
-    initialComponentHasOptionList,
-  );
+export function EditTab({ component, handleComponentChange }: EditTabProps): React.ReactElement {
+  const componentHasOptionList: boolean = !!component.optionsId || !!component.options;
   const errorMessage = useComponentErrorMessage(component);
+  const previousComponent = usePrevious(component);
+  const dialogRef = useRef<HTMLDialogElement>(null);
+
+  useUpdate(() => {
+    if (!previousComponent.options && !!component.options) dialogRef.current.showModal();
+  }, [component, previousComponent]);
 
   return (
     <>
       {componentHasOptionList ? (
-        <SelectedOptionList
-          setComponentHasOptionList={setComponentHasOptionList}
+        <OptionListEditor
+          ref={dialogRef}
           component={component}
           handleComponentChange={handleComponentChange}
         />
       ) : (
-        <div className={classes.optionButtons}>
-          <AddManualOptionsModal
-            setComponentHasOptionList={setComponentHasOptionList}
-            component={component}
-            handleComponentChange={handleComponentChange}
-          />
-          <OptionListSelector
-            setComponentHasOptionList={setComponentHasOptionList}
-            component={component}
-            handleComponentChange={handleComponentChange}
-          />
-          <OptionListUploader
-            setComponentHasOptionList={setComponentHasOptionList}
-            component={component}
-            handleComponentChange={handleComponentChange}
-          />
+        <div className={classes.container}>
+          <AddOptionList component={component} handleComponentChange={handleComponentChange} />
         </div>
       )}
-      {errorMessage && (
-        <StudioErrorMessage className={classes.errorMessage} size='small'>
-          {errorMessage}
-        </StudioErrorMessage>
-      )}
+      {errorMessage && <StudioErrorMessage size='small'>{errorMessage}</StudioErrorMessage>}
     </>
   );
 }
 
-type SelectedOptionListProps = {
-  setComponentHasOptionList: (value: boolean) => void;
-} & Pick<IGenericEditComponent<SelectionComponentType>, 'component' | 'handleComponentChange'>;
+type AddOptionListProps = EditTabProps;
 
-function SelectedOptionList({
-  setComponentHasOptionList,
-  component,
-  handleComponentChange,
-}: SelectedOptionListProps) {
+function AddOptionList({ component, handleComponentChange }: AddOptionListProps) {
   const { t } = useTranslation();
 
-  const handleDelete = () => {
-    if (component.options) {
-      delete component.options;
-    }
-
-    const emptyOptionsId = '';
-    handleComponentChange({
-      ...component,
-      optionsId: emptyOptionsId,
-    });
-
-    setComponentHasOptionList(false);
+  const handleInitialManualOptionsChange = () => {
+    handleOptionsChange({ component, handleComponentChange, options: [] });
```

Comment:
The object passed to the handle-function should be a new object created from a function that is defined outside the component. 
```suggestion
    const updatedComponent = updateComponentWithInitialManualOptions(component);
    handleOptionsChange(updatedComponent);
```